### PR TITLE
Fboemer/bool ops2

### PR DESCRIPTION
**Pull request by fboemer**
* Expands use of bitwise operations to signed / unsigned integers
* Fixes some inconsistencies between op names ("bit_and" => "and", etc.), which was causing errors

### CODE REVIEW

File: plaidml2/edsl/edsl_test.cc [hunk 1]
```diff
@@ -83,6 +83,125 @@ TEST(CppEdsl, Cast) {
...
+  }
+}
+
+TEST(CppEdsl, BitXor) {
+  auto A = Placeholder(PLAIDML_DATA_UINT64, {2, 2});
+  auto B = Placeholder(PLAIDML_DATA_UINT64, {2, 2});
+  auto C = A ^ B;
+  Program program("bit_xor", {C});
+
+  std::vector<std::uint64_t> input_a{1, 2, 3,  //
```
Comment (jbruestle) [inline plaidml2/edsl/edsl_test.cc]: Looks like this is being used as a 2x2 value, so we probably shouldn't have a 3x3 buffer for clarity.

File: pmlc/dialect/stripe/into_mlir.cc [hunk 2]
```diff
@@ -212,6 +212,7 @@ static void IntrinsicIntoMLIR(OpBuilder* builder, SymbolTable* locals, const str
     throw std::runtime_error("Unknown intrinsic: " + intrinsic.name);
   }
   auto genericBuilder = abstractOp->getInterface<util::GenericBuilder>();
+  IVLOG(1, "genericBuilder");
```
Comment (jbruestle) [inline pmlc/dialect/stripe/into_mlir.cc]: It's OK to add IVLOG messages, but level 1 is usually reserved for messages that might of of interest to most users, and debug level logs should be lowered to 3 or 4 for commit.

File: pmlc/dialect/eltwise/ir/predicates.td [hunk 3]
```diff
@@ -54,6 +54,10 @@ def AnyScalarUnsigned : Type<And<[
     CPred<"is_uint($_self.cast<ScalarType>().type())">
   ]>, "scalar-unsigned">;
 
+def AnyScalarUnsignedOrInteger : Type<Or<[
+  AnyScalarInteger.predicate, AnyScalarUnsigned.predicate
```
Comment (flaub) [inline pmlc/dialect/eltwise/ir/predicates.td]: Indent this please

File: pmlc/dialect/stripe/into_mlir.cc (hunk 2 above)
Comment (flaub) [inline pmlc/dialect/stripe/into_mlir.cc]: Still needed?

File: pmlc/dialect/eltwise/ir/ops.td [hunk 4]
```diff
@@ -136,25 +136,26 @@ class EW_CompareOp<string mnemonic, list<OpTrait> traits = [NoSideEffect]> :
...
+    Arguments<(ins EltwiseUnsignedOrInteger:$lhs, EltwiseUnsignedOrInteger:$rhs, TypeAttr:$type)>,
+    Results<(outs EltwiseUnsignedOrInteger:$result)>;
+
 
 def EW_AbsOp : EW_UnaryOp<"abs">;
 def EW_ACosOp : EW_UnaryFloatOp<"acos">;
 def EW_AddOp : EW_BinaryOp<"add", [Commutative, NoSideEffect]>, HasFolder;
-def EW_AndOp : EW_BinaryIntOp<"and", [Commutative, NoSideEffect]>;
+def EW_BitAndOp : EW_BinaryOp<"bit_and", [Commutative, NoSideEffect]>;
+def EW_BitNotOp : EW_UnaryOp<"bit_not">;
```
Comment (flaub) [inline pmlc/dialect/eltwise/ir/ops.td]: Wouldn't we want to restrict this to integer types?

File: pmlc/dialect/eltwise/ir/ops.td [hunk 5]
```diff
@@ -136,25 +136,26 @@ class EW_CompareOp<string mnemonic, list<OpTrait> traits = [NoSideEffect]> :
...
+def EW_BitOrOp : EW_BinaryOp<"bit_or", [Commutative, NoSideEffect]>;
+def EW_BitXorOp : EW_BinaryOp<"bit_xor", [Commutative, NoSideEffect]>;
```
Comment (flaub) [inline pmlc/dialect/eltwise/ir/ops.td]: It seems like we should restrict this to integer types.

File: pmlc/dialect/stripe/into_mlir.cc (hunk 2 above)
Comment (fboemer) [reply pmlc/dialect/stripe/into_mlir.cc]: Removed

**Pull request merged**